libs/stats/odc/stats/tasks.py

```diff
@@ -55,16 +58,35 @@ def sanitize(v):
         if isinstance(v, datetime):
             return v.isoformat()
         return v
+
     return transform_object_tree(sanitize, query)
 
 
+def download_from_s3(cache):
+    s3 = boto3.resource("s3")
+    path_parts = cache.replace("s3://", "").split("/")
+    bucket = path_parts.pop(0)
+    key = "/".join(path_parts)
```

Comment:
There are common functions all over the place for getting a bucket and key from a s3 URI...

Like: https://github.com/opendatacube/odc-tools/blob/develop/libs/aws/odc/aws/__init__.py#L35

Reply:
I've moved this to odc.aws and rely on an existing function 's3_url_parse'.